app/code/core/Mage/CatalogRule/Model/Resource/Rule.php

```diff
@@ -206,7 +206,7 @@ public function insertRuleData(Mage_CatalogRule_Model_Rule $rule, array $website
         $toTime = $toTime ? ($toTime + self::SECONDS_IN_DAY - 1) : 0;
 
         $timestamp = time();
-        if ($fromTime > $timestamp
+        if ($fromTime > $timestamp + self::SECONDS_IN_DAY
             || ($toTime && $toTime < $timestamp)
         ) {
```

Comment:
```suggestion
        if ($fromTime > strtotime('+1day', $timestamp) || ($toTime && $toTime < $timestamp)) {
```
his shouldn't create problems with daylight saving time change etc